dali/kernels/imgproc/convolution/cutlass/threadblock/predicated_tile_iterator.h

```diff
@@ -369,20 +546,320 @@ class PredicatedTileIterator<Shape_, Element_, layout::PitchLinear, AdvanceRank,
+                                                                  aligned_offset.aligned_element);
+    mux(dst, access_ptr[0], access_ptr[1], aligned_offset.offset);
+  }
+
+  /**
+   * @brief Load and add kernel elements from window_element - Inner Conv variant
+   *
+   * In inner conv, we skip first and last row for border handling,
+   * as the loads are row-wise, this can be a no-op if mask_first or mask_last is true.
+   */
+  template <bool mirrored = true, bool IsInnerConv_ = kIsInnerConv>
+  CUTLASS_DEVICE std::enable_if_t<IsInnerConv_> add_vec(AccessType &dst, int window_element,
+                                                      bool mask_first, bool mask_last) {
+    static_assert(mirrored, "All accesses should be mirrored for inner conv.");
+    if (mask_first || mask_last) {
+      return;
+    }
+    auto aligned_offset = get_aligned_offset<mirrored>(window_element);
+    constexpr int window_center = ConvWindowConfiguration::template getWindowCenter<mirrored>();
+    const auto *access_ptr = reinterpret_cast<AccessType const *>(pointer_ + window_center +
+                                                                  aligned_offset.aligned_element);
+    mux_add(dst, access_ptr[0], access_ptr[1], aligned_offset.offset);
+  }
+
+  /**
+   * @brief Load and add kernel elements from window_element - Outer Conv variant
+   *
+   * In inner conv, we skip first and last column for border handling,
+   * so we sometimes need to mask the addition of first of last vector element.
+   * As for "vector aligned" loads, the data must be a multpile of that size, we only do it
+   * for the first or last element.
+   */
+  template <bool mirrored, bool IsInnerConv_ = kIsInnerConv>
+  CUTLASS_DEVICE std::enable_if_t<!IsInnerConv_> add_vec(AccessType &dst, int window_element,
+                                                       bool mask_first, bool mask_last) {
+    // In outer conv we skip columns
+    int lo_offset = get_lo_offset<mirrored>(window_element);
+    int offset = get_offset<mirrored>(window_element, lo_offset);
+    constexpr int window_center = ConvWindowConfiguration::template getWindowCenter<mirrored>();
+    const auto *access_ptr =
+        reinterpret_cast<AccessType const *>(pointer_ + window_center + lo_offset);
+    Element tmp = static_cast<Element>(0);
+
+    if (mask_first) {
+      tmp = static_cast<Element>(dst[0]);
+    } else if (mask_last) {
+      tmp = static_cast<Element>(dst[AccessSize - 1]);
+    }
+
+    mux_add(dst, access_ptr[0], access_ptr[1], offset);
+
+    if (mask_first) {
+      dst[0] = static_cast<Element>(tmp);
+    } else if (mask_last) {
+      dst[AccessSize - 1] = static_cast<Element>(tmp);
+    }
+  }
+
+  /**
+   * @brief Channel information is necessary only for inner convolution, for outer it's always 1
+   */
+  CUTLASS_DEVICE
+  int Channels() {
+    if (kIsInnerConv) {
+      switch (channels_) {
```

Comment:
@mzient Would that help the compiler?

Reply:
I checked. It doesn't help; this code is simply eliminated.